src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/AttachExceptionRecordToExistingCaseTest.java

```diff
@@ -95,34 +93,61 @@ public void setup() throws InterruptedException, ServiceBusException, JSONExcept
             .atMost(60, TimeUnit.SECONDS)
             .pollDelay(2, TimeUnit.SECONDS)
             .until(() -> isExceptionRecordCreated(randomPoBox));
+    }
+
+    @Test
+    public void should_atatch_exception_record_to_the_existing_case() throws JSONException {
+        //given
+        Envelope newEnvelope = updateEnvelope("envelopes/new-envelope.json");
+        caseDetails = ccdCaseCreator.createCase(newEnvelope);
+
+        // when
+        callCallbackUrlToAttachExceptionToCase();
 
+        //then
+        await("Exception record is attached to the case")
+            .atMost(60, TimeUnit.SECONDS)
+            .pollDelay(2, TimeUnit.SECONDS)
+            .until(this::isExceptionRecordAttachedToTheCase);
+        verifyExistingCaseIsUpdatedWithExceptionRecordData();
     }
 
     @Test
-    @Disabled
-    public void should_atatch_exception_record_to_the_existing_case() {
+    public void should_atatch_exception_record_to_the_existing_case_with_evidence_documents() throws JSONException {
+        //given
+        Envelope newEnvelope = updateEnvelope("envelopes/new-envelope-with-evidence.json");
+        caseDetails = ccdCaseCreator.createCase(newEnvelope);
 
         // when
+        callCallbackUrlToAttachExceptionToCase();
+
+        //then
+        await("Exception record is attached to the case")
+            .atMost(60, TimeUnit.SECONDS)
+            .pollDelay(2, TimeUnit.SECONDS)
+            .until(this::isExceptionRecordAttachedToTheCase);
+
+        verifyExistingCaseIsUpdatedWithExceptionRecordData();
+    }
+
+    public void callCallbackUrlToAttachExceptionToCase() {
+        Map<String, Object> caseData = exceptionRecord.getData();
+        caseData.put("attachToCaseReference", String.valueOf(caseDetails.getId()));
+
         CallbackRequest callbackRequest = CallbackRequest
             .builder()
-            .eventId("attach_case")
-            .caseDetails(exceptionRecordCase)
-            .caseDetailsBefore(existingCase)
+            .eventId("attachToExistingCase")
+            .caseDetails(exceptionRecord)
             .build();
 
-        Response callbackResponse = RestAssured
+        RestAssured
```

Comment:
So im still not convinced this is the right approach for a "Functional test"
The integration tests are comprehensive and do exactly this.

I still strongly believe we should be poking ccd to make the callback to orchestrator.
This the the only code path that is not tested in our automated testing and needs to be done.
and this is the bit that is the easiest to break too.

Reply:
@vakepati

Reply:
You still need this functional test as you are just mocking in the integration which doesn't assures that it will update case in CCD when call-back endpoint is invoked. If you have concerns around this please take it on slack

Reply:
@nitinprabhu No doubt we need the interactions in this test so having this is better than not but we should also be testing the actual callback which can easily be broken ( I broke it in the demo for instance)

Reply:
that would be contract testing. Callback is invoked as part of click of submit event for attach to existing case.

Reply:
If you mess with the definition callback will be broken any ways

Reply:
hmmm not sure I agree ... contract testing would be similar to our integration test and this.
Im dont believe that contract testing would actually cover this due to the dynamic nature of the spreadsheets so likelyhood is that this particular call would never properly fall under those sets of tests.

Anyway there is still a gap of testing the actual callback from ccd in the environment before the code goes to production.
I think that there should be a pipeline for publishing the spreadsheet that kicks off an orchestrator build that checks that it can still interoperate with ccd. 
This is why I think this should be included in our functional tests to ensure that it does not break and go live broken.

Obviously that is my opinion from my experience in CD ... so it should be up to QA to decide if the current test/s are good enough for CD.